File: since2020/CodeForces/1327B.py
'''
	@judge CodeForces
	@id 1327B
	@name Princesses and Princes
	@contest Educational Codeforces Round 84

	@tag Greedy, Graph Theory
'''
from sys import stdin

def solve(graph):
	m = max(map(lambda g: max(g) if len(g) else 0, graph))
	prince   = set(range(1, m + 2))
	princess = set()
	for idx, edges in enumerate(graph):
		target = prince & edges
		if len(target) == 0:
			princess |= set([ idx + 1 ])
		else:
			prince -= set([ min(target) ])
	if len(princess) == 0:
		print('OPTIMAL')
	else:
		print('IMPROVE')
		print('{} {}'.format(min(princess), min(prince)))
	return

input()
for line in stdin:
	n = int(line)
	graph = [ set(list(map(int, stdin.readline().split()))[1:]) for _ in range(n) ]
	solve(graph)

File: since2020/CodeForces/1327B.cpp
/**
 * 	@judge CodeForces
 *  @id 1327B
 *  @name Princesses and Princes
 *  @contest Educational Codeforces Round 84
 * 
 *  @tag Greedy, Graph Theory
 */
#include <bits/stdc++.h>
using namespace std;

typedef long long int LL;

int n, p;
vector<vector<int>> g;
vector<int> pk, ps;

void solve() {
	pk = vector<int>(p + 1, 1);
	ps = vector<int>(n, 1);
	for (int i = 0; i < n; i++)
		for (const auto & x: g[i])
			if (pk[x]) {
				pk[x] = ps[i] = 0;
				break;
			}
	int a1, a2;
	for (a1 = 0; a1 < n; a1++)
		if (ps[a1])
			break;
	for (a2 = 1; a2 <= p; a2++)
		if (pk[a2])
			break;
	if (a1 == n) {
		cout << "OPTIMAL" << endl;
		return ;
	}
	cout << "IMPROVE" << endl;
	cout << a1 + 1 << " " << a2 << endl;
}

int main() {
	ios::sync_with_stdio(false);
	cin.tie(nullptr);
	cout.tie(nullptr);
	int t, k;
	for (cin >> t; t; t--) {
		cin >> n;
		g = vector<vector<int>>(n);
		p = 0;
		for (auto && v: g) {
			cin >> k;
			v = vector<int>(k);
			for (auto && x: v) {
				cin >> x;
				p = max(p, x);
			}
		}
		solve();
		g.clear();
	}
}